### PR TITLE
Support apispec 1.0.0

### PR DESCRIPTION
**Pull request by DStape**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
     'flask>=0.10.1',
     'marshmallow>=2.0.0',
     'webargs>=0.18.0',
-    'apispec>=0.17.0',
+    'apispec==1.0.0',
```
Comment (Jamim) [inline setup.py]: Hello @DStape,
Why is it pinned to `1.0.0`?
I suppose there should be `apispec>=1.0.0` instead.

Comment (DStape) [reply setup.py]: Good catch, will fix just now...

File: tests/test_extension.py [hunk 2]
```diff
@@ -92,9 +92,9 @@ def test_serve_swagger_ui_custom_url(self, app, client):
     def test_apispec_config(self, app):
         app.config['APISPEC_TITLE'] = 'test-extension'
         app.config['APISPEC_VERSION'] = '2.1'
+        app.config['APISPEC_OA_VERSION'] = '2.0'
```
Comment (sloria) [inline tests/test_extension.py]: How about `APISPEC_OAS_VERSION`, for consistency with the [official abbreviation](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#introduction)?

File: tests/test_openapi.py [hunk 3]
```diff
@@ -13,13 +13,14 @@
 
 @pytest.fixture()
 def marshmallow_plugin():
-    return MarshmallowPlugin()
+    return MarshmallowPlugin(schema_name_resolver=lambda x: None)
```
Comment (sloria) [inline tests/test_openapi.py]: Is the `schema_name_resolver` argument necessary?

Comment (DStape) [reply tests/test_openapi.py]: It's not 100% necessary, however as of apispec v1.0.0rc1, the response schema will be placed in the definitions section of the spec and then referenced from the appropriate place - meaning some of the ```test_responses``` tests  ```tests/test_openapi.py``` will fail, e.g.:

```
def test_responses(self, schemas, path, openapi):
        response = path['get']['responses']['default']
        assert response['description'] == 'a band'
        expected = openapi.schema2jsonschema(schemas.BandSchema)
>       assert response['schema'] == expected
E       AssertionError: assert {'$ref': '#/definitions/Band'} == {'properties':...pe': 'object'}
E         Left contains more items:
E         {u'$ref': u'#/definitions/Band'}
E         Right contains more items:
E         {u'properties': {'genre': {u'type': u'string'}, 'name': {u'type': u'string'}},
E          u'type': u'object'}
E         Use -v to get the full diff

tests/test_openapi.py:83: AssertionError
```

It was simply easier to set ```schema_name_resolver``` to a ```None```-returning callable, than to change the tests, although I'm happy to do that if you think that would be more appropriate :).

File: tests/test_extension.py (hunk 2 above)
Comment (DStape) [reply tests/test_extension.py]: Agreed, best to stick with the offical abbrev., fixed.

**Comment by DStape** [comment]
@sloria Thank you for the review, merge and release!

**Pull request merged**